modules/home-manager/firefox.nix

```
@@ -66,6 +116,37 @@ let
                 theme = themes.${profile.flavor}.${profile.accent};
               };
             };
+
+            settings.${stylusExtensionId} = mkIf profile.userstyles.enable {
+              inherit (profile.userstyles) force;
+              settings =
+                let
+                  # default settings applied to all userstyles
+                  globalStyleSettings = {
+                    global = {
+                      lightFlavor = profile.userstyles.flavor;
+                      darkFlavor = profile.userstyles.flavor;
+                      accentColor = profile.userstyles.accent;
+                    };
+                  };
+
+                  # settings applied per userstyle as mentioned above
+                  # these take priority over global settings
+                  perStyleSettings = (
+                    listToAttrs (
+                      mapAttrsToList (name: value: {
+                        name = "Userstyle ${name} Catppuccin";
+                        value = mapAttrs (_: castToInt) value;
+                      }) profile.userstyles.settings
+                    )
+                  );
```

[mapAttrs'](https://noogle.dev/f/lib/mapAttrs') effectively does the same with less lines :D

```suggestion
                  perStyleSettings = lib.mapAttrs' (name: value: {
                    name = "Userstyle ${name} Catppuccin";
                    value = mapAttrs (_: castToInt) value;
                   }) profile.userstyles.settings;
```

---

That's so cool :o I need to learn about more of lib's prime functions